### PR TITLE
Consider the new f-string tokens for `flake8-commas`

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This fixes the bug where the `flake8-commas` rules weren't taking the new f-string tokens into account.

## Test Plan

Add new test cases around f-strings for all of `flake8-commas`'s rules.

fixes: #8556

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_commas/rules/trailing_commas.rs [hunk 1]
```diff
@@ -29,22 +30,33 @@ enum TokenType {
 
 /// Simplified token specialized for the task.
 #[derive(Copy, Clone)]
-struct Token<'tok> {
-    type_: TokenType,
-    // Underlying token.
-    spanned: Option<&'tok Spanned>,
+struct Token {
+    r#type: TokenType,
+    range: TextRange,
```
Comment (dhruvmanila) [inline crates/ruff_linter/src/rules/flake8_commas/rules/trailing_commas.rs:35]: The raw token was never used, so let's simplify this a bit.

I've also renamed `type_` to `r#type`. This is just a personal preference and I can revert if the former is better.

File: crates/ruff_linter/src/rules/flake8_commas/rules/trailing_commas.rs [hunk 2]
```diff
@@ -222,21 +234,49 @@ pub(crate) fn trailing_commas(
...
+                }
+            }
+            _ => {
+                if fstrings == 0 {
+                    Some(Token::from(spanned))
+                } else {
+                    None
+                }
+            }
+        });
```
Comment (dhruvmanila) [inline crates/ruff_linter/src/rules/flake8_commas/rules/trailing_commas.rs:271]: This is the main change.

File: crates/ruff_linter/src/rules/flake8_commas/rules/trailing_commas.rs [hunk 3]
```diff
@@ -222,21 +234,49 @@ pub(crate) fn trailing_commas(
...
+                fstrings = fstrings.saturating_add(1);
+                None
+            }
+            Tok::FStringEnd => {
+                fstrings = fstrings.saturating_sub(1);
+                if fstrings == 0 {
+                    indexer
+                        .fstring_ranges()
+                        .outermost(tok_range.start())
+                        .map(|range| Token::new(TokenType::String, range))
```
Comment (dhruvmanila) [inline crates/ruff_linter/src/rules/flake8_commas/rules/trailing_commas.rs:259]: We're getting the range on `FStringEnd`, so the `None` case isn't possible.

**Comment by dhruvmanila** [comment]
The ecosystem change is the fact that we've stopped checking for trailing comma within the expression part of f-strings.

**Review by charliermarsh — APPROVED** [review]
Great, thanks!

**Pull request merged**